components/calendar/src/chinese.rs

```diff
@@ -0,0 +1,1213 @@
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5273-L5281
+    pub(crate) fn minor_solar_term_from_fixed(date: RataDie) -> i32 {
+        let moment: Moment = date.as_moment();
+        let offset = Self::chinese_offset(date);
+        let universal: Moment = Location::universal_from_standard(moment, offset);
+        let solar_longitude = i64_to_i32(Astronomical::solar_longitude(universal) as i64);
+        debug_assert!(
+            matches!(solar_longitude, I32Result::WithinRange(_)),
+            "Solar longitude should be in range of i32"
+        );
+        let s = solar_longitude.saturate();
+        adjusted_rem_euclid(3 + quotient(s - 15, 30), 12)
+    }
+
+    /// Returns the UTC time offset in China based on the date. The Chinese calendar determines the month based on
+    /// observations from China, so this function is necessary to discern the time offset in China, which changed in
+    /// 1929 when China adopted a standard time zone set at 120 degrees longitude.
+    fn chinese_offset(date: RataDie) -> f64 {
+        let year = Iso::iso_from_fixed(date).year().number;
+        if year < 1929 {
+            UTC_OFFSET_PRE_1929
+        } else {
+            UTC_OFFSET_POST_1929
+        }
+    }
+
+    /// The fixed date in Chinese standard time of the next new moon on or after a given Moment.
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5329-L5338
+    fn chinese_new_moon_on_or_after(moment: Moment) -> RataDie {
+        let new_moon_moment = Astronomical::new_moon_at_or_after(Self::midnight_in_china(moment));
+        let chinese_offset = Self::chinese_offset(new_moon_moment.as_rata_die());
+        Location::standard_from_universal(new_moon_moment, chinese_offset).as_rata_die()
+    }
+
+    /// The fixed date in Chinese standard time of the previous new moon before a given Moment.
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5318-L5327
+    fn chinese_new_moon_before(moment: Moment) -> RataDie {
+        let new_moon_moment = Astronomical::new_moon_before(Self::midnight_in_china(moment));
+        let chinese_offset = Self::chinese_offset(new_moon_moment.as_rata_die());
+        Location::standard_from_universal(new_moon_moment, chinese_offset).as_rata_die()
+    }
+
+    /// Universal time of midnight at start of a Moment in China
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5353-L5357
+    fn midnight_in_china(date: Moment) -> Moment {
```

Comment:
suggestion: if `date` is of type `Moment`, then rename to `moment`, which is consistent with other similar fns